crates/core/src/client/mod.rs

```diff
@@ -216,14 +216,26 @@ impl KakarotClient for KakarotClientImpl {
         };
         // Make the function call to get the Starknet contract address
         let starknet_contract_address = self.client.call(request, starknet_block_id).await?;
-        // Concatenate the result of the function call
-        let concatenated_result = starknet_contract_address
-            .into_iter()
-            .fold(FieldElement::ZERO, |acc, x| acc + x);
+
+        if starknet_contract_address.len() != 1 {
+            return Err(KakarotClientError::OtherError(anyhow::anyhow!(
+                "Kakarot get_code: starknet_contract_address length is not 1"
+            )));
+        };
+
+        // shadow the variable to FielElement from a Vec<FieldElement>, for use in subsequent code
+        let starknet_contract_address = match starknet_contract_address.get(0) {
+            Some(x) => *x,
+            None => {
+                return Err(KakarotClientError::OtherError(anyhow::anyhow!(
```

Comment:
This feels weird because you should never reach this arm since you already checked that `starknet_contract_address.len() == 1`. Maybe something like:
```
        let starknet_contract_address = match starknet_contract_address.get(0) {
            Some(x) if starknet_contract_address.len() == 1 => *x,
            _ => {
                return Err(KakarotClientError::OtherError(anyhow::anyhow!(
                      "Kakarot get_code: starknet_contract_address length is not 1"
                )));
            }
        };
```

Reply:
Hello @greged93, I do not completely get this.

So, in match the other possible case is anyways `None`, so how is `_` better here { I see the point if there were multiple other possibilities for this value, then the above syntax feels like, "for anything else, let's throw error"  }

Is this some rust best practice to use, when other statements of match should not be reachable?

Reply:
No no, what I mean is that you checked the length of your array in the `if` condition. So you're kind of checking the length twice. I would remove the if and update with what I sent. What do you think?

Reply:
Oh! Now I get it makes sense!

Thanks, this is cleaner :)

Reply:
done [here](https://github.com/kkrt-labs/kakarot-rpc/pull/168/commits/2ee183d9fc761fa0dbf61fafed56b0641a8de6fb).

Reply:
rebased on main as well.